### PR TITLE
Expressiontool timer

### PR DESCRIPTION
**Pull request by douglowe**
This PR is in response to [issue 1680](https://github.com/common-workflow-language/cwltool/issues/1680).

1. Increased the default time for expression tools to 60 seconds, so that `--eval-timeout` flag will be used more for optimisation purposes, rather than being needed to accomodate systems with slower response times for expression tools.
2. Added some log information for situations where the expression tool time limit is triggered.

Outstanding questions:
1. The `exec_js_process` routine seems to be used for organisational tasks at the start of the workflow. And is run with a 30 second time limit. If the tool time limit gets breached for that process then my log message will be misleading (because the `--eval-timeout` flag doesn't control that time limit).
2. Really it would be useful for the workflow to exit if a tool fails in this manner. Perhaps when this is implemented the log messages could be moved to a more appropriate location in the code (solving the issue I note above)?

### CODE REVIEW

**Comment by douglowe** [comment]
3rd outstanding question (while I remember).

I've moved `timer.cancel()` and my `timer.is_alive()` check to after `stdin_buf.close()`, because I don't know if there's any risk in reading `stdout_buf` and `stderr_buf` before closing this. If closing `stdin_buf` takes any significant time to perform, then there's a risk of a false triggering of my timer check. For a longterm fix it might be better to reorganise my code here.

**Comment by mr-c** [comment]
> 1\. The `exec_js_process` routine seems to be used for organisational tasks at the start of the workflow. And is run with a 30 second time limit. If the tool time limit gets breached for that process then my log message will be misleading (because the `--eval-timeout` flag doesn't control that time limit).

Good catch. We can refactor the code to respect `--eval-timeout` here. Do you want help with that?

> 2\. Really it would be useful for the workflow to exit if a tool fails in this manner. Perhaps when this is implemented the log messages could be moved to a more appropriate location in the code (solving the issue I note above)?

Sure, you could raise a `JavascriptException` when the timeout is triggered. That should cause a reasonable failure. We'll add tests for the different scenarios to confirm that.

> 3rd outstanding question (while I remember).
> 
> I've moved `timer.cancel()` and my `timer.is_alive()` check to after `stdin_buf.close()`, because I don't know if there's any risk in reading `stdout_buf` and `stderr_buf` before closing this. If closing `stdin_buf` takes any significant time to perform, then there's a risk of a false triggering of my timer check. For a longterm fix it might be better to reorganise my code here.

Good to know. I think it will be okay for now. What do you think @tetron ?

File: cwltool/sandboxjs.py [hunk 1]
```diff
@@ -285,12 +285,20 @@ def process_finished() -> bool:
                         pipes[1].write(buf)
         except OSError:
             break
-    timer.cancel()
 
     stdin_buf.close()
-    stdoutdata = stdout_buf.getvalue()[: -len(PROCESS_FINISHED_STR) - 1]
-    stderrdata = stderr_buf.getvalue()[: -len(PROCESS_FINISHED_STR) - 1]
+
```
[the remark this answers is not in the document]
Comment (douglowe) [reply cwltool/sandboxjs.py]: Yep - I've run this and pushed the changes. I'll make sure to check using this before pushing more changes later.

**Comment by douglowe** [comment]
> > 1. The `exec_js_process` routine seems to be used for organisational tasks at the start of the workflow. And is run with a 30 second time limit. If the tool time limit gets breached for that process then my log message will be misleading (because the `--eval-timeout` flag doesn't control that time limit).
> 
> Good catch. We can refactor the code to respect `--eval-timeout` here. Do you want help with that?
> 
Yep - I'm happy to help with that.

> > 2. Really it would be useful for the workflow to exit if a tool fails in this manner. Perhaps when this is implemented the log messages could be moved to a more appropriate location in the code (solving the issue I note above)?
> 
> Sure, you could raise a `JavascriptException` when the timeout is triggered. That should cause a reasonable failure. We'll add tests for the different scenarios to confirm that.
> 
I'll have a look at adding this in.

**Comment by douglowe** [comment]
I've refactored the code to use `--eval-timeout` to control `exec_js_process` within the `validate_js_expressions` step. This has involved duplicating `eval_timeout` to be stored in `LoadingContext`, mirroring `RuntimeContext`.

**Pull request merged**